src/components/Buttons/AccountSelectButton/AccountSelectorButton.tsx

```diff
@@ -0,0 +1,59 @@
+import { ComponentPropsWithoutRef } from "react";
+
+export interface AccountSelectorProps extends ComponentPropsWithoutRef<"div"> {
+  theme?: string;
+  profileimage: React.ReactNode;
+  accountname: string;
+  massatoken: React.ReactNode;
+  accountbalance: string;
+  children?: React.ReactNode;
+}
+
+export function AccountSelectorButton(props: AccountSelectorProps) {
+  // Specific classes for the account selector button
+  const accountSelectorClass = "flex flex-row items-center justify-between";
+
+  const hoverClass = `hover:bg-info`;
+  const activeClass = `active:bg-secondaryAlt`;
+
+  // Generic classes
+  const genericFlex = "flex justify-center items-center";
+
+  // correct text displacement due to line height
+
+  const tokenDimensionClass = `h-6 w-6`;
+
+  return (
+    // Take more time to think about correct code structure
+    // Check grid
+    <div
+      className={` bg-neutral text-primary h-[56px] w-full p-3 rounded-lg mas-menu-active cursor-pointer 
+                      ${accountSelectorClass} 
+                      ${hoverClass} 
+                      ${activeClass}`}
+      {...props}
+      data-testid="account-selector-button"
+    >
+      <div className={`${genericFlex}`}>
+        <div
+          className={`bg-primary h-8 w-8 rounded-full mr-2
+                           ${genericFlex}`}
+        >
+          {props.profileimage}
+        </div>
+        <div>{props.accountname}</div>
+      </div>
+
```

Comment:
@mazmassa is this okay for the time ? That way once i have correctly deved the component I can integrate it here correctly ?